libbeat/processors/add_process_metadata/gosigar_cid_provider.go

```diff
@@ -106,25 +107,35 @@ func (p gosigarCidProvider) getProcessCgroups(pid int) (map[string]string, error
 // ID is found then an empty string is returned.
 // Example:
 // /kubepods/besteffort/pod9b9e44c2-00fd-11ea-95e9-080027421ddf/2bb9fd4de339e5d4f094e78bb87636004acfe53f5668104addc761fe4a93588e
-func (p gosigarCidProvider) getCid(cgroups map[string]string) string {
+func (p gosigarCidProvider) getCid(cgroups map[string]cgroup.ControllerPath) string {
 	// if regex defined use it to find cid
 	if len(p.cgroupRegex) != 0 {
 		re := regexp.MustCompile(p.cgroupRegex)
```

Comment:
No need to fix this in this PR, but if `p.cgroupRegex` comes from user configuration we shouldn't use `regexp.MustCompile`, this will panic if the user configures an invalid regex.

Reply:
Yah, true...